src/storage/DataAccessorBasedStore.ts

```diff
@@ -115,6 +118,9 @@ export class DataAccessorBasedStore implements ResourceStore {
     // In the future we want to use getNormalizedMetadata and redirect in case the identifier differs
     let metadata = await this.accessor.getMetadata(identifier);
     let representation: Representation;
+    if (conditions) {
+      this.validateConditions(true, conditions, metadata);
+    }
```

Comment:
```suggestion
    this.validateConditions(true, conditions, metadata);
```
The if-statement is not necessary since this check already happens in `validateConditions` (and removing it makes it consistent with the calls in the other functions).